### PR TITLE
Fix video wrapper

### PR DESCRIPTION
**Pull request by patrickab**
The goal is to correcly render the videos produced by the RenderImageInfoWrapper. Currently only sequences of black images are produced

### CODE REVIEW

File: prefq/examples/imitation_preference_comparisons.py [hunk 1]
```diff
@@ -5,27 +5,194 @@
 # Disable formatting and linters so we can keep the example as close to the original as possible for now.
 # fmt: off
 # pylint: skip-file
+
+import time
+import uuid
```
Comment (timokau) [inline prefq/examples/imitation_preference_comparisons.py:10]: This import seems unused.

File: prefq/examples/imitation_preference_comparisons.py [hunk 2]
```diff
@@ -5,27 +5,194 @@
...
+import json
+import os
+import pathlib
+import tempfile
+from collections import defaultdict
```
Comment (timokau) [inline prefq/examples/imitation_preference_comparisons.py:15]: Unused import.

File: prefq/examples/imitation_preference_comparisons.py [hunk 3]
```diff
@@ -5,27 +5,194 @@
...
+    SynchronousHumanGatherer,
+    write_fragment_video,
+)
+from imitation.data.types import TrajectoryWithRewPair
+from imitation.data.wrappers import RenderImageInfoWrapper
 from imitation.policies.base import FeedForward32Policy, NormalizeFeaturesExtractor
 from imitation.rewards.reward_nets import BasicRewardNet
 from imitation.rewards.reward_wrapper import RewardVecEnvWrapper
+from imitation.util import logger as imit_logger
+from imitation.util import video_wrapper
```
Comment (timokau) [inline prefq/examples/imitation_preference_comparisons.py:31]: Unused import.

File: prefq/examples/imitation_preference_comparisons.py [hunk 4]
```diff
@@ -5,27 +5,194 @@
...
 from imitation.util.networks import RunningNorm
 from imitation.util.util import make_vec_env
 from stable_baselines3 import PPO
 from stable_baselines3.common.evaluation import evaluate_policy
 from stable_baselines3.ppo import MlpPolicy
 
+
+class PrefqGatherer(SynchronousHumanGatherer):
```
Comment (timokau) [inline prefq/examples/imitation_preference_comparisons.py:39]: Please add a comment that clarifies that this is a slightly modified version of the `PrefCollectGatherer` from https://github.com/HumanCompatibleAI/imitation/pull/716.

File: prefq/examples/imitation_preference_comparisons.py [hunk 5]
```diff
@@ -5,27 +5,194 @@
...
+                    else: 
+                        print("Query Client: Feedback received")
+                        break
+            return feedback_data
+
+        feedback_data = _wait_for_feedback_request(self.server_url + "feedback")
+        return feedback_data
+
+
+SERVER_URL = "http://127.0.0.1:5000/"
```
Comment (timokau) [inline prefq/examples/imitation_preference_comparisons.py:178]: Please move constants to the top of the file.

File: prefq/examples/imitation_preference_comparisons.py [hunk 6]
```diff
@@ -5,27 +5,194 @@
...
+
 rng = np.random.default_rng(0)
+video_dir = tempfile.mkdtemp(prefix="videos_")
+print(os.path.dirname(os.path.abspath(__file__)))
```
Comment (timokau) [inline prefq/examples/imitation_preference_comparisons.py:182]: Should we keep this print? If so, please be a bit more verbose to clarify what is being printed to make the logs more readable.

File: prefq/examples/imitation_preference_comparisons.py [hunk 7]
```diff
@@ -5,27 +5,194 @@
...
 
-venv = make_vec_env("Pendulum-v1", rng=rng)
+venv = make_vec_env(env_name = "Pendulum-v1", 
+                    rng=rng,
+                    post_wrappers=[lambda env, env_id: RenderImageInfoWrapper(env)],
+)
+
+#RenderImageInfoWrapper(env=venv, scale_factor=0.5, use_file_cache=True)
```
Comment (timokau) [inline prefq/examples/imitation_preference_comparisons.py:189]: Remove this.

File: prefq/examples/imitation_preference_comparisons.py [hunk 8]
```diff
@@ -42,6 +209,11 @@
     ),
     env=venv,
     n_steps=2048 // venv.num_envs,
+    seed=0,
+    batch_size=64,
+    ent_coef=0.0,
+    learning_rate=0.0003,
+    n_epochs=10,
```
Comment (timokau) [inline prefq/examples/imitation_preference_comparisons.py:216]: Why these changes?

File: prefq/examples/imitation_preference_comparisons.py [hunk 9]
```diff
@@ -61,8 +235,13 @@
     preference_querent=querent,
     preference_gatherer=gatherer,
     reward_trainer=reward_trainer,
+    fragment_length=100,
+    transition_oversampling=1,
+    initial_comparison_frac=0.1,
+    allow_variable_horizon=False,
```
Comment (timokau) [inline prefq/examples/imitation_preference_comparisons.py:241]: Why these changes?

File: prefq/imitation_server.py [hunk 10]
```diff
@@ -0,0 +1,184 @@
+"""A webserver that asks the user preference queries."""
```
Comment (timokau) [inline prefq/imitation_server.py:1]: Is this a leftover or why do we need a separate server?

File: scripts/run_imitation_server.sh [hunk 11]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
```
Comment (timokau) [inline scripts/run_imitation_server.sh:1]: Same here: Leftover / why necessary?

**Review by timokau — CHANGES_REQUESTED** [review]
Thanks! I left some review comments. In addition, please make sure that the pre-commit checks pass.

File: prefq/examples/imitation_preference_comparisons.py (hunk 6 above)
Comment (patrickab) [reply prefq/examples/imitation_preference_comparisons.py:182]: just a relict, just removed it

File: prefq/examples/imitation_preference_comparisons.py (hunk 8 above)
Comment (patrickab) [reply prefq/examples/imitation_preference_comparisons.py:216]: these changes have been copied from one of the examples provided by the authors of the PrefCollectGatherer: 

https://github.com/HumanCompatibleAI/imitation/blob/4dbf20c6d42286377901dd538cd6c5001225fa40/docs/tutorials/5b_train_preference_comparisons_with_synchronous_human_feedback.ipynb

File: prefq/examples/imitation_preference_comparisons.py (hunk 9 above)
Comment (patrickab) [reply prefq/examples/imitation_preference_comparisons.py:241]: these changes have been copied from one of the examples provided by the authors of the PrefCollectGatherer: 

https://github.com/HumanCompatibleAI/imitation/blob/4dbf20c6d42286377901dd538cd6c5001225fa40/docs/tutorials/5b_train_preference_comparisons_with_synchronous_human_feedback.ipynb

File: prefq/imitation_server.py (hunk 10 above)
Comment (patrickab) [reply prefq/imitation_server.py:1]: due to differences in implementation, several changes in have to be made. eg, the default flask server (server.py) communicates with the QueryClient (query_client.py), that can request and receive stored data from the server at any given moment, while the imitation server (imiation_server.py) is supposed to send stored feedback to the PrefqGatherer (defined in imitation_preference_comparisons.py) only when all videos have been evalutated.

File: scripts/run_imitation_server.sh (hunk 11 above)
Comment (patrickab) [reply scripts/run_imitation_server.sh:1]: This line of code has been copied from (https://github.com/patrickab/prefq/blob/main/scripts/run_imitation_example.sh) without further consideration

File: prefq/examples/imitation_preference_comparisons.py (hunk 8 above)
Comment (timokau) [reply prefq/examples/imitation_preference_comparisons.py:211]: I think those changes should not be relevant for the imitation integration.

File: scripts/run_imitation_server.sh (hunk 11 above)
Comment (timokau) [reply scripts/run_imitation_server.sh]: I think those changes should not be relevant for the imitation integration.

File: prefq/imitation_server.py (hunk 10 above)
Comment (timokau) [reply prefq/imitation_server.py:1]: Can't we use the query client api here as well?

**Comment by patrickab** [comment]
rebased all commits to a new branch called "integrate-imitation"
